tests/MC/RISCV/XCVmac_valid_riscv32_xcvmac.txt.yaml

```
@@ -0,0 +1,560 @@
+
+  -
+    input:
+      bytes: [ 0xdb, 0x52, 0x03, 0xfe ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulhhurn t0, t1, zero, 31"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x42, 0x73, 0x00 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulsn t0, t1, t2, 0"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x42, 0x03, 0x00 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulsn t0, t1, zero, 0"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x42, 0x03, 0x20 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulsn t0, t1, zero, 16"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x42, 0x03, 0x3e ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulsn t0, t1, zero, 31"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x42, 0x73, 0x80 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulsrn t0, t1, t2, 0"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x42, 0x03, 0xa0 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulsrn t0, t1, zero, 16"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x42, 0x03, 0xbe ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulsrn t0, t1, zero, 31"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x52, 0x73, 0x00 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulun t0, t1, t2, 0"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x52, 0x03, 0x00 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulun t0, t1, zero, 0"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x52, 0x03, 0x20 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulun t0, t1, zero, 16"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x52, 0x03, 0x3e ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulun t0, t1, zero, 31"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x52, 0x73, 0x80 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
+    expected:
+      insns:
+        -
+          asm_text: "cv.mulurn t0, t1, t2, 0"
+
+  -
+    input:
+      bytes: [ 0xdb, 0x52, 0x03, 0xa0 ]
+      arch: "CS_ARCH_RISCV"
+      options: [ "riscv32", "xcvmac" ]
```

```suggestion
      options: [ "riscv32", "xcvmac" ]
```

These `riscv32` should be mapped to `CS_MODE_32`